pydomcfg/utils.py

```diff
@@ -103,84 +110,62 @@ def _smooth_MB06(depth: DataArray, rmax: float) -> DataArray:
-    zri = np.ones(zenv.shape)  # * zrmax
-    zrj = np.ones(zenv.shape)  # * zrmax
-
-    tol = 1.0e-8
-    itr = 0
-    max_itr = 10000
-
-    while itr <= max_itr and (zrmax - rmax) > tol:
-
-        itr += 1
-        zrmax = 0.0
-        # we set zrmax from previous r-values (zri and zrj) first
-        # if set after current r-value calculation (as previously)
-        # we could exit DO WHILE prematurely before checking r-value
-        # of current zenv
-        max_zri = np.nanmax(np.absolute(zri))
-        max_zrj = np.nanmax(np.absolute(zrj))
-        zrmax = np.nanmax([zrmax, max_zrj, max_zri])
-
-        print("Iter:", itr, "rmax: ", zrmax)
-
-        zri *= 0.0
-        zrj *= 0.0
-
-        for j in range(nj - 1):
-            for i in range(ni - 1):
-                ip1 = np.minimum(i + 1, ni)
-                jp1 = np.minimum(j + 1, nj)
-                if zenv[j, i] > 0.0 and zenv[j, ip1] > 0.0:
-                    zri[j, i] = (zenv[j, ip1] - zenv[j, i]) / (
-                        zenv[j, ip1] + zenv[j, i]
-                    )
-                if zenv[j, i] > 0.0 and zenv[jp1, i] > 0.0:
-                    zrj[j, i] = (zenv[jp1, i] - zenv[j, i]) / (
-                        zenv[jp1, i] + zenv[j, i]
-                    )
-                if zri[j, i] > rmax:
-                    ztmpi1[j, i] = zenv[j, ip1] * zrfact
-                if zri[j, i] < -rmax:
-                    ztmpi2[j, ip1] = zenv[j, i] * zrfact
-                if zrj[j, i] > rmax:
-                    ztmpj1[j, i] = zenv[jp1, i] * zrfact
-                if zrj[j, i] < -rmax:
-                    ztmpj2[jp1, i] = zenv[j, i] * zrfact
-
-        ztmpi = np.maximum(ztmpi1, ztmpi2)
-        ztmpj = np.maximum(ztmpj1, ztmpj2)
-        zenv = np.maximum(zenv, np.maximum(ztmpi, ztmpj))
-
-    da_zenv.data = zenv
-    return da_zenv
+    # Initialize envelope bathymetry
+    zenv = depth
+
+    for _ in range(max_iter):
+
+        # Initialize lists of DataArrays to concatenate
+        all_ztmp = []
+        all_zr = []
+        for dim in zenv.dims:
+
+            # Shifted arrays
+            zenv_m1 = zenv.shift({dim: -1})
+            zenv_p1 = zenv.shift({dim: +1})
+
+            # Compute zr
+            mb06 = (zenv_m1 - zenv) / (zenv_m1 + zenv)
+            zr = mb06.where((zenv > 0) & (zenv_m1 > 0), 0)
+            for dim_name in zenv.dims:
+                zr[{dim_name: -1}] = 0
+            all_zr += [zr]
+
+            # Compute ztmp
+            zr_p1 = zr.shift({dim: +1})
+            all_ztmp += [zenv.where(zr <= rmax, zenv_m1 * zrfact)]
+            all_ztmp += [zenv.where(zr_p1 >= -rmax, zenv_p1 * zrfact)]
+
+        # Update envelope bathymetry
+        zenv = xr.concat([zenv] + all_ztmp, "dummy_dim").max("dummy_dim")
+
+        # Check target rmax
+        zr = xr.concat(all_zr, "dummy_dim")
+        if ((np.abs(zr) - rmax) <= tol).all():
+            return zenv
+
+    # TODO:
+    #   Warning or error?
+    warnings.warn(
```

Comment:
I personally would put an error: from personal experience, usually it is sufficient tiincrease the tolerance of a couple of orders, an many tines it is relating with the precision of your data ... in this way we force the user to be aware and maybe make the input more consistent precision wise

Reply:
OK, changed to error! Maybe we should decrease the default number of iterations... I'd get pretty mad if I wait hours to get an error and no return :) 
Anyways, not urgent.

Reply:
Yes, I agree!